cpp/include/nvtext/generate_ngrams.hpp

```diff
@@ -47,18 +47,20 @@ namespace nvtext {
  * @throw cudf::logic_error if `separator` is invalid
  * @throw cudf::logic_error if there are not enough strings to generate any ngrams
  *
- * @param strings Strings column to tokenize and produce ngrams from.
+ * @param input Strings column to tokenize and produce ngrams from
  * @param ngrams The ngram number to generate.
  *               Default is 2 = bigram.
  * @param separator The string to use for separating ngram tokens.
  *                  Default is "_" character.
- * @param mr Device memory resource used to allocate the returned column's device memory.
- * @return New strings columns of tokens.
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @return New strings columns of tokens
  */
 std::unique_ptr<cudf::column> generate_ngrams(
-  cudf::strings_column_view const& strings,
+  cudf::strings_column_view const& input,
   cudf::size_type ngrams               = 2,
-  cudf::string_scalar const& separator = cudf::string_scalar{"_"},
+  cudf::string_scalar const& separator = cudf::string_scalar{"_", true, cudf::get_default_stream()},
```

Comment:
Note that in many cases we might avoid a race if the input stream is not a non-blocking stream, IOW if it synchronizes with the CUDA default stream and cudf continues to use the CUDA default stream as its own default. However, in that case using the default stream here would trigger an implicit synchronization, which is also suboptimal.

Reply:
I think for the functions in this PR, I can remove the default parameters.
I'm not sure if this approach will always work in other APIs but I'm comfortable removing them at least for these.

Reply:
Yeah I suppose if you can just remove them here we can cross the bigger bridge when we come to it. I can also bring it up with the team so that there's awareness.